gridt/src/app/core/settings.service.ts

```diff
@@ -22,26 +24,97 @@ export class SettingsService {
     SETIDFAIL: "Could not SET identity",
     NOTLOGGEDIN: "Not logged in",
     SECSTOREUNAVAILABLE: "Secure storage not available.",
-  }
+  };
+
+  public empty_identity: Identity = {
+    id: null,
+    username: "",
+    bio: "",
+    email: "",
+    avatar: ""
+  };
+
+  /**
+  * Emits the last known identity
+  */
+  private _user_identity$ = new ReplaySubject<Identity>(1);
+
+  /**
+  * This Observable can be used in applications to display the latest known
+  * identity of the user.
+  */
+  public userIdentity$: Observable<Identity> = this._user_identity$.asObservable();
+
+  /**
+  * This function updates the user identity.
+  * - If both server and local identity are unavailable, emit empty ID.
+  * - When server ID is retreived, but local ID has not been set yet, store
+  *   server ID in local and emit local ID.
+  * - When the server ID is different from local ID, overwrite local ID with
+  *   the server ID. Then emit local ID.
+  * - If both the server and local identity agree, emit local ID.
+  * - If the server is unavailable but local ID is available, emit local ID.
+  * - Only when updateIdentity() function is called should the userIdentity$
+  *   change.
+  *
+  * - Makes sure all errors are caught such that this function always gives an
+  *   output.
+  */
+  private mapErrorToEmptyID = pipe(
+    catchError( err => {console.warn(err); return of<Identity>(this.empty_identity);}),
+    flatMap( (id) => of(<Identity>(id)))
+  );
+
+  private mapErrorToFalse = pipe(
+    catchError( err => {console.warn(err); return of<boolean>(false);})
+  );
+
+  public updateIdentity(): void {
```

Comment:
I'm on my phone right now but first impression is that this is really well thought out and I cant find any failures. Also for something this complicated it is super great to have this many comments.